src/ftp.rs

```diff
@@ -58,38 +58,47 @@ impl FtpStream {
     /// Switch to secure mode if possible. If the connection is already
     /// secure does nothing.
     ///
+    /// ## Panics
+    ///
+    /// Panics if the plain TCP connection cannot be switched to TLS mode.
+    ///
     /// ## Example
     ///
     /// ```
     /// use ftp::FtpStream;
     /// let mut ftp_stream = FtpStream::connect("127.0.0.1:21").unwrap();
     /// // Switch to the secure mode
-    /// let mut ftp_stream = ftp_stream.secure().unwrap();
+    /// let (mut ftp_stream, _) = ftp_stream.secure();
     /// // Do all secret things
     /// let _ = ftp_stream.quit();
     /// ```
     ///
     #[cfg(feature = "secure")]
-    pub fn secure(mut self) -> Result<FtpStream> {
+    pub fn secure(mut self) -> (FtpStream, Result<()>) {
         let secured = self.reader.get_ref().is_ssl();
         if secured {
-            Ok(self)
+            (self, Ok(()))
         }
         else {
             // Ask the server to start securing data
             let auth_command = String::from("AUTH TLS\r\n");
-            try!(self.write_str(&auth_command));
-            try!(self.read_response(status::AUTH_OK));
+            if let Err(e) = self.write_str(&auth_command) {
+                return (self, Err(e));
+            }
+
+            if let Err(e) = self.read_response(status::AUTH_OK) {
+                return (self, Err(e));
+            }
 
             // Initialize SSL and make the opened stream secured
             let ssl = match Ssl::new(&SSL_CONTEXT) {
                 Ok(ssl) => ssl,
-                Err(e) => return Err(Error::new(ErrorKind::Other, e))
+                Err(e) => return (self, Err(Error::new(ErrorKind::Other, e)))
             };
 
             let stream = match SslStream::connect(ssl, self.reader.into_inner().into_tcp_stream()) {
                 Ok(stream) => stream,
-                Err(e) => return Err(Error::new(ErrorKind::Other, e))
+                Err(e) => panic!("error: cannot open SSL connection: {}", e)
```

Comment:
Well if we can't recover from it then we should panic. I would say to also panic on 96 because we can't recover from unpredictable behavior